query/parse_test.go

```diff
@@ -64,7 +64,7 @@ func TestParseQuery(t *testing.T) {
 		{"abccase:yes", &Substring{Pattern: "abccase:yes"}},
 		{"file:abc", &Substring{Pattern: "abc", FileName: true}},
 		{"branch:pqr", &Branch{Pattern: "pqr"}},
-		{"((x) )", &Regexp{Regexp: mustParseRE("(x)")}},
+		{"((x) )", &Substring{Pattern: "x"}},
```

Comment:
it looks like the intent here was to actually get a regex out. Can we instead change the input so that our optimizer doesn't remove Substring for this case (But keep it in others). EG using a `(x|y)` as the pattern.